src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -7174,7 +7187,51 @@ public void execute()
 
       abstract void doExtract(final JsArrayString response);
    }
+
+   private void openFileFromMessage(final String file,
+                                    final int line,
+                                    final int column)
+   {
+
+      FilePosition filePosition = FilePosition.create(line, column);
+      CodeNavigationTarget navigationTarget = new CodeNavigationTarget(file, filePosition);
+
+      fileTypeRegistry_.editFile(
+         FileSystemItem.createFile(navigationTarget.getFile()),
+         filePosition);
+   }
+
+   public static void onOpenFileFromMessage(final String file, int line, int column)
+   {
+      if (activeTextEditingTarget_ != null)
```

Comment:
seems like we could use `isActiveDocument()` here?

Reply:
Moved all this to the viewer pane, since otherwise, we won't be able to open files when all files are open.